charts/kube-prometheus-stack/templates/prometheus-operator/deployment.yaml

```diff
@@ -39,7 +39,8 @@ spec:
           {{- end }}
           imagePullPolicy: "{{ .Values.prometheusOperator.image.pullPolicy }}"
           args:
-            {{- if semverCompare "< v0.39.0" .Values.prometheusOperator.image.tag }}
+            {{- if eq .Values.prometheusOperator.image.tag "master" -}}
```

Comment:
@rothsa thanks I wasn't aware there was a `master` tag.

But just so I understand the issue, what happens when you configure `prometheusOperator.image.tag` to `master`? I haven't tested it but guessing `semverCompare` thinks `master` is `<` `v0.39.0`? Assuming yes (or whatever the reason), would you also add a comment above this line explaining why there's an empty `if` condition? I can easily imagine this being cleaned up in future if there's no why.

Just a thought - since I see there are multiple `master*` tags (`master-arm64` etc), it may be better to use [regexMatch ](http://masterminds.github.io/sprig/strings.html#regexmatch-mustregexmatch) check instead of checking only `master`. WDYT?

Reply:
@scottrigby Correct, it throws an error when trying to compare `master` to a semver. Here's what we were seeing when trying to set the `prometheusOperator.image.tag` to `master`:

`Error: UPGRADE FAILED: template: kube-prometheus-stack/templates/prometheus-operator/deployment.yaml:42:19: executing "kube-prometheus-stack/templates/prometheus-operator/deployment.yaml" at <semverCompare "< v0.39.0" .Values.prometheusOperator.image.tag>: error calling semverCompare: Invalid Semantic Version`

I'll let @rothsa answer the regex question but shouldn't be too hard to implement a `master*` regex.

Reply:
Sounds fair 🙂

Reply:
Looks like `{{- if regexMatch "master.*" .Values.prometheusOperator.image.tag -}}` does the trick. I'll touch base with @rothsa this morning.

Reply:
Yes, this looks great. Metroshica is correct about the error. Thanks for the suggestion and the changes.